ubuntu-advantage-client/ec2_cleanup.py

```diff
@@ -19,62 +19,125 @@ def parse_args():
         help=("Tag used to filter cloud resources for deletion. "
               "Default: {}".format(CI_DEFAULT_TAG))
     )
+    parser.add_argument(
+        "-o", "--older-than", dest="older_than", action="store",
+        help=("Tag used to filter cloud resources for deletion. "
+              "Format: MM/DD/YY [HH:MM:SS].)"
+    )
     return parser.parse_args()
 
 
-def clean_ec2(tag):
+def get_tag_prefix(older_than):
+    if not older_than:
+       return ""
+    print('Match only resources created before %s' % older_than)
+    try:
+        time = datetime.datetime.strptime(older_than, "%m/%d/%y %H:%M:%S")
+        return time.strftime("uaclient-ci-%m%d-%H%M%S")
+    except ValueError:
+        time = datetime.datetime.strptime(older_than, "%m/%d/%y")
+        return time.strftime("uaclient-ci-%m%d")
+
+
+def is_resource_stale(resource, tag_time_prefix):
+    """Return True if the given resource is older than the tag_time_prefix.
```

Comment:
so this is "tag_time_prefix" vs the method that's just "tag_prefix". Since it's more than the time should the argument just be "tag_prefix"?

Reply:
thanks @mitechie.

sorry, this review was on an earlier push of mine. I went through a bit of bikeshedding in my head and ended up with the following:
```
def delete_resource_by_tag(resource, tag, time_prefix):                         
```
which returns a bool True if the resource should be deleted based on tag or tag_time_prefix values.

Please take a look and see if you think that satisfies the intent of your concern here